client/state/posts/reducer.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import indexBy from 'lodash/collection/indexBy';
+
+/**
+ * Internal dependencies
+ */
+import {
+	POST_RECEIVE,
+	POSTS_RECEIVE,
+	POSTS_REQUEST,
+	POSTS_REQUEST_SUCCESS,
+	POSTS_REQUEST_FAILURE
+} from 'state/action-types';
+
+/**
+ * Tracks all known post objects, indexed by post global ID.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = {}, action ) {
+	switch ( action.type ) {
+		case POST_RECEIVE:
+			state = Object.assign( {}, state, {
+				[ action.post.global_ID ]: action.post
+			} );
+			break;
+
+		case POSTS_RECEIVE:
+			state = Object.assign( {}, state, indexBy( action.posts, 'global_ID' ) );
+			break;
```

Comment:
As in my comment above, we have two code paths here for essentially the same action. Could we not simply always use the list case, even guarding it with a simple concat?

Then I think it would be clear exactly what the intended action is and also it would be clear that the result of getting one post is really no different than that of when getting ten posts or one hundred posts etc…